Maybe consider making shorter names for tests

I installed an Ubuntu 16.04 with an encrypted disk and I get the following:

```
ubuntu@test:~/openenclave/build$ cmake ..
-- The C compiler identification is Clang 7.0.1
-- The CXX compiler identification is Clang 7.0.1
-- The ASM compiler identification is Clang
-- Found assembler: /usr/bin/clang-7
-- Check for working C compiler: /usr/bin/clang-7
-- Check for working C compiler: /usr/bin/clang-7 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/clang++-7
-- Check for working CXX compiler: /usr/bin/clang++-7 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Not using CCache
-- Performing Test SPECTRE_MITIGATION_C_FLAGS_SUPPORTED
-- Performing Test SPECTRE_MITIGATION_C_FLAGS_SUPPORTED - Success
-- Performing Test SPECTRE_MITIGATION_CXX_FLAGS_SUPPORTED
-- Performing Test SPECTRE_MITIGATION_CXX_FLAGS_SUPPORTED - Success
-- Spectre 1 mitigations supported
-- Valgrind not found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
ENABLE_FULL_LIBCXX_TESTS not set - building some libcxx tests
USE_DEBUG_MALLOC is set, building oecore with memory leak detection.
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.11") found components:  doxygen dot
-- Configuring done
CMake Error: Cannot open file for write: /home/ubuntu/openenclave/build/tests/libcxx/enc/CMakeFiles/libcxxtest-std_thread_thread.mutex_thread.mutex.requirements_thread.sharedtimedmutex.requirements_thread.sharedtimedmutex.class_lock_shared.pass_enc.dir/build.make.tmp
CMake Error: : System Error: File name too long
CMake Error: Cannot open file for write: /home/ubuntu/openenclave/build/tests/libcxx/enc/CMakeFiles/libcxxtest-std_thread_thread.mutex_thread.mutex.requirements_thread.sharedtimedmutex.requirements_thread.sharedtimedmutex.class_lock_shared.pass_enc.dir/depend.make.tmp
CMake Error: : System Error: File name too long
CMake Error: Cannot open file for write: /home/ubuntu/openenclave/build/tests/libcxx/enc/CMakeFiles/libcxxtest-std_thread_thread.mutex_thread.mutex.requirements_thread.sharedtimedmutex.requirements_thread.sharedtimedmutex.class_lock_shared.pass_enc.dir/flags.make.tmp
CMake Error: : System Error: File name too long
CMake Error: Cannot open file for write: /home/ubuntu/openenclave/build/tests/libcxx/enc/CMakeFiles/libcxxtest-std_thread_thread.mutex_thread.mutex.requirements_thread.sharedtimedmutex.requirements_thread.sharedtimedmutex.class_lock_shared.pass_enc.dir/link.txt.tmp
CMake Error: : System Error: File name too long
CMake Error: Could not create /home/ubuntu/openenclave/build/tests/libcxx/enc/CMakeFiles/libcxxtest-std_thread_thread.mutex_thread.mutex.requirements_thread.sharedtimedmutex.requirements_thread.sharedtimedmutex.class_lock_shared.pass_enc.dir/cmake_clean.cmake
CMake Error: Cannot open file for write: /home/ubuntu/openenclave/build/tests/libcxx/enc/CMakeFiles/libcxxtest-std_thread_thread.mutex_thread.mutex.requirements_thread.sharedtimedmutex.requirements_thread.sharedtimedmutex.class_lock_shared.pass_enc.dir/DependInfo.cmake.tmp
CMake Error: : System Error: File name too long
-- Generating done
CMake Error: Cannot open file for write: /home/ubuntu/openenclave/build/tests/libcxx/enc/CMakeFiles/libcxxtest-std_thread_thread.mutex_thread.mutex.requirements_thread.sharedtimedmutex.requirements_thread.sharedtimedmutex.class_lock_shared.pass_enc.dir/progress.make.tmp
CMake Error: : System Error: File name too long
-- Build files have been written to: /home/ubuntu/openenclave/build
ubuntu@test:~/openenclave/build$ getconf PATH_MAX .
4096
ubuntu@test:~/openenclave/build$ getconf NAME_MAX .
143
ubuntu@test:~/openenclave/build$ uname -a
Linux test 4.4.0-143-generic #169-Ubuntu SMP Thu Feb 7 07:56:38 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
ubuntu@test:~/openenclave/build$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.6 LTS
Release:        16.04
Codename:       xenial
ubuntu@test:~/openenclave/build$
```

I'm not sure if we should mark this one as a bug or not, but maybe we want to shorten those filenames in the future.

CC: @andschwa , @johnkord

--- a/cmake/get_testcase_name.cmake
+++ b/cmake/get_testcase_name.cmake
@@ -19,5 +19,16 @@ function(get_testcase_name FILENAME NAMEVAR PREFIX)
         endif()
         string(REGEX REPLACE "[/=]" "_" n ${n})
         string(REGEX REPLACE "[\!]" "-" n ${n})
+
+        # Encrypted filesystems on Linux have a smaller limit on the
+	# maximum file name length. Limit the testcase name to the
+	# last 120 characters.
+	set(max_length 120)
+	string(LENGTH ${n} n_length)
+	if (${n_length} GREATER ${max_length})
+	    math(EXPR start_pos "${n_length} - ${max_length}")
+	    string(SUBSTRING ${n} ${start_pos} ${max_length} n)
+	endif()
+
 	set(${NAMEVAR} ${n} PARENT_SCOPE)
 endfunction(get_testcase_name)

--- a/cmake/get_testcase_name.cmake
+++ b/cmake/get_testcase_name.cmake
@@ -13,22 +13,22 @@
 # namevar - variable to receive the testcase name
 #
 function(get_testcase_name FILENAME NAMEVAR PREFIX)
-        string(REGEX REPLACE "\.c(pp)?$" "" n ${FILENAME})
-        if (NOT ${PREFIX} STREQUAL "")
-            string(REGEX REPLACE ${PREFIX} "" n ${n})
-        endif()
-        string(REGEX REPLACE "[/=]" "_" n ${n})
-        string(REGEX REPLACE "[\!]" "-" n ${n})
+  string(REGEX REPLACE "\.c(pp)?$" "" n ${FILENAME})
+  if (NOT PREFIX STREQUAL "")
+    string(REGEX REPLACE ${PREFIX} "" n ${n})
+  endif ()
+  string(REGEX REPLACE "[/=]" "_" n ${n})
+  string(REGEX REPLACE "[\!]" "-" n ${n})
 
-        # Encrypted filesystems on Linux have a smaller limit on the
-	# maximum file name length. Limit the testcase name to the
-	# last 120 characters.
-	set(max_length 120)
-	string(LENGTH ${n} n_length)
-	if (${n_length} GREATER ${max_length})
-	    math(EXPR start_pos "${n_length} - ${max_length}")
-	    string(SUBSTRING ${n} ${start_pos} ${max_length} n)
-	endif()
+  # Encrypted filesystems on Linux have a smaller limit on the
+  # maximum file name length. Limit the testcase name to the
+  # last 120 characters.
+  set(max_length 120)
+  string(LENGTH ${n} n_length)
+  if (n_length GREATER max_length)
+    math(EXPR start_pos "${n_length} - ${max_length}")
+    string(SUBSTRING ${n} ${start_pos} ${max_length} n)
+  endif ()
 
-	set(${NAMEVAR} ${n} PARENT_SCOPE)
+  set(${NAMEVAR} ${n} PARENT_SCOPE)
 endfunction(get_testcase_name)

